receiver/memoryreceiver/monitor.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memoryreceiver
+
+import (
+	"context"
+	"runtime"
+	"time"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/shirou/gopsutil/mem"
+	"go.uber.org/zap"
+
+	"github.com/Omnition/omnition-opentelemetry-collector/ptypes/ptime"
+)
+
+// Monitor encapsulates the core logic of scraping and exporting memory metrics of the host
+type Monitor struct {
+	logger    *zap.Logger
+	resource  *resourcepb.Resource
+	startTime time.Time
+	stopCh    chan struct{}
+
+	nextConsumer consumer.MetricsConsumer
+}
+
+// Start starts the memory monitor and scrapes metrics after time interval provided in a loop
+func (m *Monitor) Start(interval time.Duration) {
+	m.stopCh = make(chan struct{})
+	for {
+		select {
+		case <-m.stopCh:
+			return
+		case <-time.After(interval):
+			m.scrapeAndEmit()
+		}
+	}
+}
+
+// Stop stops the memory monitor
+func (m *Monitor) Stop() {
+	close(m.stopCh)
+}
+
+func (m *Monitor) scrapeAndEmit() {
+	ctx := context.Background()
+	metrics := make([]*metricspb.Metric, 0, len(allMetrics))
+
+	// mem.VirtualMemory is a gopsutil function
+	memInfo, err := mem.VirtualMemory()
+	if err != nil {
+		if err == context.DeadlineExceeded {
+			m.logger.Debug("unable to collect memory time info", zap.Error(err))
+		} else {
+			m.logger.Error("unable to collect momory time info", zap.Error(err))
```

Comment:
```suggestion
			m.logger.Error("unable to collect memory time info", zap.Error(err))
```
Is there a way to nicely combine Error/Debug calls and eliminate `if`?